### PR TITLE
 Nerfs xenobio's ability to lag the server

### PR DESCRIPTION
**Pull request by ike709**
Optimizes slime AI process.

## Changelog
:cl:
tweak: Optimized slime processing code.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/slime/life.dm [hunk 1]
```diff
@@ -28,28 +28,31 @@
...
 				if(!Target.client || prob(20))
 					Feedon(Target)
 					AIproc = 0
 					return
-			if(attack_cooldown < world.time && (attacked || rabid))
+			if(attacked || rabid)
 				Target.attack_slime(src)
 				attack_cooldown = world.time + attack_cooldown_time
-		else if(Target in view(7, src))
+		else if((view_tracker % 3 != 0) || (Target in view(7, src))) // Only bother to check if target is still in view every third time
```
Comment (ike709) [inline code/modules/mob/living/simple_animal/slime/life.dm]: I'm fairly certain there's a faster alternative to `view()` for this usecase but I can't for the life of me remember the proc name.

**Comment by ike709** [comment]
Someone experienced with xenobio needs to test this.

**Comment by ike709** [comment]
I'm not 100% sure it works as intended but generally xenobio doesn't seem broken.

File: code/modules/mob/living/simple_animal/slime/life.dm (hunk 1 above)
Comment (Mick1299) [reply code/modules/mob/living/simple_animal/slime/life.dm]: Might be thinking of can_see?
`(can_see(Target, src, 7)`

**Comment by optimumtact** [comment]
dude I have a pro tip
```
if(prob(80)) return
```

File: code/modules/mob/living/simple_animal/slime/life.dm (hunk 1 above)
Comment (ike709) [reply code/modules/mob/living/simple_animal/slime/life.dm]: `can_see()` is horribly inaccurate.

**Comment by Kmc2000** [comment]
I believe your original issue was caused by:

/mob/living/simple_animal/slime/proc/AIprocess()
	AIproc = 1

	while(AIproc)

If you don't put a stoplag() after your while loops, they take the large sip of untracked tick time (To my knowledge, could be wrong!) and cause the game to chug immensely as that while loop gets super high priority on the thread.

**Review by PowerfulBacon — DISMISSED** [review]
Seems good in TM

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**